distribution/pull_v2.go

```diff
@@ -665,6 +665,11 @@ func (p *v2Puller) pullSchema2Layers(ctx context.Context, target distribution.De
 	}
 
 	if len(descriptors) != len(configRootFS.DiffIDs) {
+		logrus.Errorf("RootFSMismatch: len(descriptors) = %v, len(configRootFS.DiffIDs) = %v",
+			len(descriptors), len(configRootFS.DiffIDs))
+		logrus.Errorf("descriptors = %#v", descriptors)
+		logrus.Errorf("configRootFS = %#v", configRootFS)
```

Comment:
how about making this a single line? printing those values like that will probably make them very hard to parse like that, but it keeps all the information together, which should be useful when looking at the full journal output of a busy device

```suggestion
		logrus.WithFields(logrus.Fields{
			"len(descriptors)": len(descriptors),
			"len(configRootFS.DiffIDs)": len(configRootFS.DiffIDs),
			"descriptors": fmt.Sprintf("%#v", descriptors),
			"configRootFS": fmt.Sprintf("%#v", configRootFS),
		}).Error("rootFS mismatch")
```

Reply:
Right, good point, thanks. I updated the branch!